src/confighttp.cpp

```diff
@@ -551,6 +693,16 @@ namespace confighttp {
     std::stringstream configStream;
     ss << request->content.rdbuf();
     pt::ptree outputTree;
+    pt::ptree inputTree;
+    try {
+      pt::read_json(ss, inputTree);
+    }
+    catch (std::exception &e) {
+      BOOST_LOG(warning) << "UploadCover: "sv << e.what();
+      bad_request(response, request, e.what());
+      return;
+    }
+
     auto g = util::fail_guard([&]() {
```

Comment:
Perhaps this one also should be removed by calling `bad_request` where the `outputTree.put("error", "...");` is used instead? Also sending a normal response in the end.